src/main/java/pl/edu/mimuw/BankClient.java

```diff
@@ -1,35 +1,101 @@
 package pl.edu.mimuw;
 
+import java.util.ArrayList;
 import java.util.List;
 
+
 public class BankClient {
+  private final String name, surname;
+  private final String accountNumber;
+  private final int age;
+  private final List<BankAction> actionHistory;
+
+  public BankClient(BankClient client) {
+    this.name = client.name;
+    this.surname = client.surname;
+    this.age = client.age;
+    this.accountNumber = client.accountNumber;
+    this.actionHistory = new ArrayList<>();
 
-  // TODO:
-  // - create proper constructor
-  // - implement methods
-  // - add your own methods (1 is enough)
+    for (BankAction ba : client.actionHistory) {
+      this.actionHistory.add(ba.getCopy());
+    }
+  }
+
+  public BankClient(String name, String surname, int age, String accountNumber) {
+    this.name = name;
+    this.surname = surname;
+    this.age = age;
+    this.accountNumber = accountNumber;
+    this.actionHistory = new ArrayList<>();
+  }
 
   public String getName() {
-    throw new IllegalStateException("TODO");
+    return this.name;
   }
 
   public String getSurname() {
-    throw new IllegalStateException("TODO");
+    return this.surname;
   }
 
   public int getAge() {
-    throw new IllegalStateException("TODO");
+    return this.age;
   }
 
   public String getAccountNumber() {
-    throw new IllegalStateException("TODO");
+    return this.accountNumber;
   }
 
   public List<BankAction> getActionHistory() {
-    throw new IllegalStateException("TODO");
+    List<BankAction> l = new ArrayList<>();
+
+    for (BankAction ba : this.actionHistory) {
+      l.add(ba.getCopy());
+    }
+
+    return l;
+  }
+
+  public double getDepositsAmount() {
+    double sum = 0;
+    for (BankAction ba : this.actionHistory) {
+      if (ba instanceof DepositBankAction) {
```

Comment:
raczej nie chcemy używac instanceof w ramach programowania obiektoweg - pomysl, jak mozna by zmienic zestaw metod np w action, zeby nie bylo to konieczne, a zebys mogl rzeczywiscie miec taka metode jak `getDepositsAmount` lub może nazwana jakoś bardziej ogólnie, zeby nie była powiazana konkretnie z depozytem

Reply:
poza tym warto to jednak sumowac w dodawaniu akcji a nie tak wielokrotnie liczyc